packages/service-library/src/servicelib/aiohttp/profiler_middleware.py

```diff
@@ -0,0 +1,42 @@
+from aiohttp.web import Request, StreamResponse, middleware
+from pyinstrument import Profiler
+from servicelib.aiohttp import status
+
+from .._utils_profiling_middleware import append_profile
+
+
+def create_profiling_middleware(app_name: str):
+    @middleware
+    async def profiling_middleware(request: Request, handler):
+        profiler: Profiler | None = None
+        if request.headers.get(f"x-profile-{app_name}") is not None:
+            profiler = Profiler(async_mode="enabled")
+            profiler.start()
+
+        response = await handler(request)
+
+        if profiler is None:
+            return response
+        if response.content_type != "application/json":
+            return StreamResponse(
```

Comment:
TIP: When it comes to error responses in aiohttp you can use the corresponding exception `web.HTTPInternalError`. Note that these exceptgions are **also** inherited from `Response`. SE https://docs.aiohttp.org/en/latest/web_exceptions.html

Reply:
and you should raise them

Reply:
Thanks, very good point. This has been corrected